### DISCUSSION TITLE
Cannot set RTL in custom content language

### DISCUSSION THREAD

**Discussion by crystalenka** [joomla/joomla-cms#47092] [2. Cold Features]
### Is your feature request related to a problem? Please describe.

Some sites may need to be translated into languages that do not have language packs available. The way to manage this is to add a new content language manually, which is fine—unless the content language happens to be RTL, as there is no place to define this.

### Describe the solution you'd like

An option in the content language interface to determine whether a language is LTR or RTL.

### Additional context

This information is not currently available through content languages at all, only through installed languages.

**Comment by brianteeman** [comment]
I would not do it that way. Although the content may be us8ing RTL characters the UI will not be RTL and you will not be able to use the language switcher or set the direction in the html tag as all of this comes from the site language.

All you need to do is to create a folder for the language and a single  langmetdata.xml file inside it where you make the important definitions. (rtl, firsdat, locale, weekend, calendar). You can then discover install the language

**Comment by crystalenka** [comment]
I may be misunderstanding the purpose of being able to manually add content languages then, because it seems that I can't use them without having an installable file. I assumed (I guess incorrectly) that one would not have to create a metadata file for the purpose of adding content in that language and having it show up in the language switcher etc.

If that's the case then what is the point of adding content languages? (Genuine question, as I don't know.)

**Comment by brianteeman** [comment]
> If that's the case then what is the point of adding content languages? (Genuine question, as I don't know.)

Because a certain person in the past insisted that you should be able to have a content language and build a multilingual site without having the language installed BEFORE you went ahead and installed the language files. Don't ask me why they insissted on that facility. It never really worked in the past and definitely causes confusion as you have found.

**Comment by brianteeman** [comment]
There are several valid use case for being able to edit the content language and I guess the code could be updated to include all the variables that are in the langmetdata.xml but then you would also have to update all the references. And this is pretty much an edge case.
